### DISCUSSION TITLE
existing wireguard config

### DISCUSSION THREAD

**Discussion by lingej** [NHAS/wag#109] [Q&A]
Hello NHAS and thank you for developing wag

I am already using wireguard and have a working configuration including preshared keys
The config is already distributed to the users via Ansible and is used productively.

Is it possible to use this configuration with wag and thus only extend it with 2FA?

Thanks for your help and best regards

Joerg

**Comment by NHAS** [comment]
Howdy, 

You can't import existing devices into wag unfortunately. 

Feel free to open an issue for it though, as I think being able to do this seems reasonable

**Reply by lingej** [reply]
Hi NHAS, thanks for your quick reply.

Suppose I would analyze the sqlite database and import my current configuration. wag would then use an identical wireguard configuration.

I have not done this yet, but would this be a possible way?

If a new device is added via "wag registration --add ... ", then it would be cool if you could pass private_key, public_key and preshared_key

**Reply by NHAS** [reply]
Actually I had right forgotten that the registration URL will actually allow you to specify a public key: 

E.g
`curl https://your.wag.server/register_device?key=<token>&pubkey=<existing pub key>`

Obviously you'll still need to generate registration tokens for your various devices. And it doesnt take a preshared key unfortunately. 

But yes that is one way you could do it!


As for sqlite, wag in version 8 (which you're on if you're using the docker container) has moved to using ETCd instead so that we can do clustering. And you can definitely work out the format from there, but you shouldnt have to.

**Reply by lingej** [reply]
sqlite support is completely removed with version 8.x.x ?

**Reply by NHAS** [reply]
Not entirely. It will do a migration to etcd the first time you run it. 

But it will be later
